src/Features/CSharp/Portable/ConvertToRecord/PositionalParameterInfo.cs

```diff
@@ -3,41 +3,87 @@
             ImmutableArray<PropertyDeclarationSyntax> properties,
             INamedTypeSymbol type,
             SemanticModel semanticModel,
             CancellationToken cancellationToken)
         {
-            // first get all property symbols
+            using var _ = ArrayBuilder<PositionalParameterInfo>.GetInstance(out var resultBuilder);
+
+            // get all declared property symbols, put inherited property symbols first
             var symbols = properties
                 .SelectAsArray(p => (IPropertySymbol)semanticModel.GetRequiredDeclaredSymbol(p, cancellationToken));
 
+            // add inherited properties from a potential base record first
+            var inheritedProperties = GetInheritedPositionalParams(type, cancellationToken);
+            resultBuilder.AddRange(inheritedProperties.Select(property => new PositionalParameterInfo(
+                Declaration: null,
+                property,
+                KeepAsOverride: false)));
+
             // The user may not know about init or be converting code from before init was introduced.
             // In this case we can convert set properties to init ones
             var allowSetToInitConversion = !symbols
                 .Any(symbol => symbol.SetMethod is IMethodSymbol { IsInitOnly: true });
 
-            return properties.ZipAsArray(symbols, (syntax, symbol)
+            resultBuilder.AddRange(properties.Zip(symbols, (syntax, symbol)
                 => ShouldConvertProperty(syntax, symbol, type) switch
                 {
                     ConvertStatus.DoNotConvert => null,
-                    ConvertStatus.Override => new PositionalParameterInfo(syntax, symbol, true),
+                    ConvertStatus.Override
+                        => new PositionalParameterInfo(syntax, symbol, KeepAsOverride: true),
                     ConvertStatus.OverrideIfConvertingSetToInit
                         => new PositionalParameterInfo(syntax, symbol, !allowSetToInitConversion),
-                    ConvertStatus.AlwaysConvert => new PositionalParameterInfo(syntax, symbol, false),
+                    ConvertStatus.AlwaysConvert
+                        => new PositionalParameterInfo(syntax, symbol, KeepAsOverride: false),
                     _ => throw ExceptionUtilities.Unreachable,
-                }).WhereNotNull().AsImmutable();
+                }).WhereNotNull());
+
+            return resultBuilder.ToImmutable();
+        }
+
+        public static ImmutableArray<IPropertySymbol> GetInheritedPositionalParams(
+            INamedTypeSymbol currentType,
+            CancellationToken cancellationToken)
+        {
+            var baseType = currentType.BaseType;
+            if (baseType != null && baseType.TryGetRecordPrimaryConstructor(out var basePrimary))
+            {
+                return basePrimary.Parameters
+                    .SelectAsArray(param => param.GetAssociatedSynthesizedRecordProperty(cancellationToken)!);
```

Comment:
Why is the ! safe?

Reply:
We already know that the method is the primary record constructor, so the only way this is null is if some syntax reference call failed along the way. I changed it to use WhereNotNull instead to be safe, although if we fail here, it might produce some errors in the refactoring code.